### ISSUE TITLE
Provide docstrings

### ISSUE THREAD

**Issue by yeesian** [visr/GDAL.jl#5]
Given that we automatically check null for some of the wrapped functions, and provide some of our own types

**Comment by yeesian** [comment]
After my experience writing documentation for RasterIO, we'll probably want to automate bits of this by scraping from gdal's documentation before modifying them into docstrings.

**Comment by visr** [comment]
I'm working on automating this, after working on it a bit more I'll open a PR.

Right now I've taken the Doxyfile from the GDAL SVN, set `GENERATE_XML = YES`, then combined all resulting files in one 20MB XML. With this I can lookup the function name in the XML and add the part of the documentation I want during the rewriting. For now simply the brief description:

```julia
functionnode(et, fname) = et["/doxygen/compounddef/sectiondef/memberdef[name='$fname']"]
briefdesc = fnode["briefdescription/para/text()"]

"Find out if we have update permission for this band. "
function GDALGetRasterAccess(arg1::GDALRasterBandH)
    ccall((:GDALGetRasterAccess,libgdal),GDALAccess,(GDALRasterBandH,),arg1)
end
```

It needs more work to nicely display more extensive documentation in markdown, like this: 
http://gdal.org/gdal_8h.html#a194a45dd5319e8f0631d74a9faa518da

**Comment by yeesian** [comment]
\o/ That will be great!

**Comment by yeesian** [comment]
closed by #11
